Patient Registry System/PatientRegistrySystem.DB/Repos/DbGenericRepository/DbGeneric.cs

```diff
@@ -0,0 +1,249 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using PatientRegistrySystem.DB.Contexts;
+using PatientRegistrySystem.DB.Models;
+using PatientRegistrySystem.DB.Models.DbModels;
+using PatientRegistrySystem.Domain.Dto;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatientRegistrySystem.DB.Repos.DbGenericRepository
+{
+    public class DbGeneric<DbEntity, DomainEntity> : IDbGeneric<DbEntity, DomainEntity> where DbEntity : class, IDbModel where DomainEntity : IDomainModel
+    {
+        private readonly ApplicationIdentityDbContext _applicationIdentityDbContext;
+        private readonly IMapper _mapper;
+        private DbSet<DbEntity> _entities;
+
+        public DbGeneric(ApplicationIdentityDbContext applicationIdentityDbContext, IMapper mapper)
+        {
+            _entities = applicationIdentityDbContext.Set<DbEntity>();
+            _applicationIdentityDbContext = applicationIdentityDbContext;
+            _mapper = mapper;
+        }
+
+        public async Task<bool> CreateEntityAsync(DomainEntity entity)
+        {
+            if (entity.GetType().Equals(typeof(Record))
```

Comment:
we don't need to add these conditions, this would defeat the idea of having a generic class. Just skip doing any check, and always AddAsync. There are the generic methods that should be working for every DB entity. 

Please update this here and in the other functions as well